assign2/index.js

```diff
@@ -0,0 +1,302 @@
+    },
+  ];
+
+  let ADD_BALANCE = 0;
+  let MODAL_INMONEY = 0;
+  let MODAL_OUTMONEY = 0;
+  newList.forEach((data) => {
+    const { money, getto } = data;
+    ADD_BALANCE += getto ? money : -money;
+    let TOTAL_BALANCE = Number(ADD_BALANCE) + Number(INIT_BALANCE);
+    totalMoney.innerText = TOTAL_BALANCE.toLocaleString();
+
+    getto ? (MODAL_INMONEY += money) : (MODAL_OUTMONEY += money);
+    let TOTAL_INMONEY = Number(IN_MONEY) + Number(MODAL_INMONEY);
+    let TOTAL_OUTMONEY = Number(OUT_MONEY) + Number(MODAL_OUTMONEY);
+    plusMoney.innerText = TOTAL_INMONEY.toLocaleString();
+    minusMoney.innerText = TOTAL_OUTMONEY.toLocaleString();
+  });
+
+  //심화 2.a
+  if (howMuchSummaried.length == 0 || summarySummarized.length == 0) {
+    alert("모두 입력해주세요");
+  } else {
+    makeList(newList, list);
+    alert("저장되었습니다.");
+  }
+}
+
+saveBtn.addEventListener("click", () => addList());
+
+//5.e
+const closeModalBtn = document.getElementById("close_button");
+closeModalBtn.addEventListener("click", () => {
+  modal.classList.replace("modalStyle", HIDDEN_CLASS);
+  modalBg.classList.add(HIDDEN_CLASS);
+});
+
+//심화 1번 + 기본 4번
+const deleteButton = document.querySelectorAll(".money_and_btn > button");
+
+const deleteModal = document.getElementById("delte_modal_wrapper");
+const deleteListButton = document.getElementById("delete_confirm");
+const closeDeleteModalButton = document.getElementById("not_delete_confirm");
+
+deleteButton.forEach((button) => {
+  button.addEventListener("click", (e) => {
+    deleteModal.classList.replace(HIDDEN_CLASS, "delete_modal");
+    const deleteConfirmList = e.target.classList[0];
+    deleteListButton.addEventListener("click", () => {
+      deleteList(deleteConfirmList);
+    });
+  });
+});
+
+function deleteList(deletedId) {
+  const lists = document.querySelectorAll(".list_wrapper");
+  lists.forEach((list) => {
+    const plusMoney = document.getElementById("plus_money");
+    const minusMoney = document.getElementById("minus_money");
+
+    if (list.classList.contains(deletedId)) {
+      const h3 = list.querySelector("h3");
+      const deletedMoney = parseInt(h3.innerText.replace(/,/g, ""), 10);
+
+      if (h3.classList.contains("plus")) {
+        plusMoney.innerText = (
+          parseInt(plusMoney.innerText.replace(/,/g, ""), 10) - deletedMoney
+        ).toLocaleString();
+
+        totalMoney.innerText = (
+          parseInt(totalMoney.innerText.replace(/,/g, ""), 10) - deletedMoney
+        ).toLocaleString();
+      } else if (h3.classList.contains("minus")) {
+        minusMoney.innerText = (
+          parseInt(minusMoney.innerText.replace(/,/g, ""), 10) + deletedMoney
+        ).toLocaleString();
+
+        totalMoney.innerText = (
+          parseInt(totalMoney.innerText.replace(/,/g, ""), 10) - deletedMoney
+        ).toLocaleString();
+      }
+      list.remove();
+      deleteModal.classList.replace("delete_modal", HIDDEN_CLASS);
+    }
```

Comment:
같은 부분이 여러 번 반복되고 있어서 함수화 시키면 코드 중복을 줄일 수도 있고 가독성이 좋아질 것 같아!!!

```suggestion
   function updateMoney(element, moneyChange) {
  const moneyValue = parseInt(element.innerText.replace(/,/g, ""), 10);
  const newMoneyValue = moneyValue + moneyChange;
  element.innerText = newMoneyValue.toLocaleString();
}

if (h3.classList.contains("plus")) {
  updateMoney(plusMoney, -deletedMoney);
  updateMoney(totalMoney, -deletedMoney);
} else if (h3.classList.contains("minus")) {
  updateMoney(minusMoney, deletedMoney);
  updateMoney(totalMoney, -deletedMoney);
}

```

이런식으로 ㅎㅎ